src/components/Article.jsx

```diff
@@ -0,0 +1,100 @@
+import { useParams } from "react-router-dom";
+import CommentList from "./CommentList";
+import HeaderElement from "./HeaderElement";
+import Vote from "./Vote";
+import { getArticle } from "../utils/api";
+import { useContext, useEffect, useState } from "react";
+import { TopicsContext } from "../contexts/Topics";
+import ErrorComponent from "./ErrorComponent";
+
+export default function Article() {
+  const backButtonStyle = {
+    gridArea: "top-left-button",
+    border: "red solid 1px",
+  };
+  const homeButtonStyle = {
+    gridArea: "top-right-button",
+    textAlign: "right",
+    border: "red solid 1px",
+  };
+  const headlineStyle = { gridArea: "headline" };
+  const topicStyle = {
+    gridArea: "topic",
+    textAlign: "right",
+  };
+  const timestampStyle = {
+    gridArea: "timestamp",
+    textAlign: "right",
+  };
+  const imageStyle = { gridArea: "image" };
+  const bodyStyle = { gridArea: "body" };
+  const authorStyle = {
+    gridArea: "author",
+    textAlign: "right",
+  };
+
+  const articleId = useParams().articleUrl.match(/(?<=-)[^-]+$/);
+  const { topics, setTopics } = useContext(TopicsContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [article, setArticle] = useState({});
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+    getArticle(articleId)
+      .then(({ author, title, body, topic, created_at, article_img_url }) => {
+        const timestamp = new Date(created_at).toDateString();
+        setArticle({
+          author,
+          title,
+          body,
+          topic: topics[topic] ?? topic,
+          timestamp,
+          article_img_url,
+        });
+        setIsLoading(false);
+      })
+      .catch((err) => setIsError(true));
+  }, []);
```

Comment:
article will already come back as an object, probably better to just setArticle to what comes back from the api call and access what you need from it, no need to destructure and set each property manually as you do here